stackit/provider_acc_test.go

```diff
@@ -0,0 +1,250 @@
+	"project_id": config.StringVariable(testutil.ProjectId),
+	"name":       config.StringVariable(fmt.Sprintf("tf-acc-prov%s", acctest.RandStringFromCharSet(3, acctest.CharSetAlphaNum))),
+}
+
+// Helper function to obtain the home directory on different systems.
+// Based on os.UserHomeDir().
+func getHomeEnvVariableName() string {
+	env := "HOME"
+	switch runtime.GOOS {
+	case "windows":
+		env = "USERPROFILE"
+	case "plan9":
+		env = "home"
+	}
+	return env
+}
+
+// create temporary home and initialize the credentials file as well
+func createTemporaryHome(createValidCredentialsFile bool, t *testing.T) string {
+	// create a temporary file
+	tempHome, err := os.MkdirTemp("", "tempHome")
+	if err != nil {
+		t.Fatalf("Failed to create temporary home directory: %v", err)
+	}
+
+	// create credentials file in temp directory
+	stackitFolder := path.Join(tempHome, ".stackit")
+	if err := os.Mkdir(stackitFolder, 0o750); err != nil {
+		t.Fatalf("Failed to create stackit folder: %v", err)
+	}
+
+	filePath := path.Join(stackitFolder, "credentials.json")
+	file, err := os.Create(filePath)
+	if err != nil {
+		t.Fatalf("Failed to create credentials file: %v", err)
+	}
+	defer func() {
+		if err := file.Close(); err != nil {
+			t.Fatalf("Error while closing the file: %v", err)
+		}
+	}()
+
+	// Define content, default = invalid token
+	token := "foo_token"
+	if createValidCredentialsFile {
+		token = testutil.GetTestProjectServiceAccountToken("")
+	}
+	content := fmt.Sprintf(`
+		{
+    		"STACKIT_SERVICE_ACCOUNT_TOKEN": "%s"
+		}`, token)
+
+	if _, err = file.WriteString(content); err != nil {
+		t.Fatalf("Error writing to file: %v", err)
+	}
+
+	return tempHome
+}
+
+// Function to overwrite the home folder
+func setTemporaryHome(tempHomePath string) {
+	env := getHomeEnvVariableName()
+	if err := os.Setenv(env, tempHomePath); err != nil {
+		fmt.Printf("Error setting temporary home directory %v", err)
+	}
+}
+
+// cleanup the temporary home and reset the environment variable
+func cleanupTemporaryHome(tempHomePath string, t *testing.T) {
+	if err := os.RemoveAll(tempHomePath); err != nil {
+		t.Fatalf("Error cleaning up temporary folder: %v", err)
+	}
+	originalHomeDir, err := os.UserHomeDir()
+	if err != nil {
+		t.Fatalf("Failed to restore home directory back to normal: %v", err)
+	}
+	// revert back to original home folder
+	env := getHomeEnvVariableName()
+	if err := os.Setenv(env, originalHomeDir); err != nil {
+		fmt.Printf("Error resetting temporary home directory %v", err)
+	}
+}
+
+func getServiceAccountToken() (string, error) {
+	token, set := os.LookupEnv("TF_ACC_TEST_PROJECT_SERVICE_ACCOUNT_TOKEN")
+	if !set || token == "" {
+		return "", fmt.Errorf("Token not set, please set TF_ACC_TEST_PROJECT_SERVICE_ACCOUNT_TOKEN to a valid token to perform tests")
+	}
+	return token, nil
+}
+
+func TestAccEnvVarTokenValid(t *testing.T) {
+	// Check if acceptance tests should be run
+	if v := os.Getenv("TF_ACC"); v == "" {
+		return
+	}
```

Comment:
```suggestion
	tfAccEnv := "TF_ACC"
	if v := os.Getenv(tfAccEnv); v == "" {
		t.Skip(fmt.Sprintf(
			"Acceptance tests skipped unless env '%s' set",
			tfAccEnv))
		return
	}
```
I think it would be better to add a `t.Skip()`, so that when `go test -v provider_acc_test.go` runs, the tests are marked as skipped tests and not passed